### PR TITLE
Send internal logs from prover and verifier

### LINKED ISSUE #13092: Send internal logs from prover and verifier to daemon
Use RPC to send internal logs from prover and verifier to make them available via the `internalLogs` query in the ITN GraphQL server.

Add optional `process` field to the logs.

### PR DESCRIPTION
**Pull request by psteckler**
Send internal logs from the prover and verifier to the daemon, where they are queued for the ITN GraphQL server along with the daemon internal logs.

When the prover and verifier are started, `set_data` sets a process kind and the port for the daemon to receive RPC requests. `Itn_logger.log` calls the RPC if the process kind has been set, otherwise it queues the log as usual. When the RPC data is received by the daemon, it calls `log` again, and this time, the log is queued on the daemon.

Tested by running a local network, verifying that logs from the daemon, prover, and verifier appear in the GraphQL data.

Log item from daemon:
```
{ 
    "id": 506,
    "process": null, 
    "message": "Generate_next_state_done",  
    "metadata": [],
    "timestamp": "2023-04-27 01:07:46.027103Z"
}
```

Log item from verifier:
```
{
    "id": 547,
    "process": "verifier",
    "message": "Verify_heterogenous",
    "metadata": [
      {
        "value": 0,
        "item": "count"
      }
    ],
    "timestamp": "2023-04-27 01:07:56.750841Z"
}
```

Log item from prover:
```
{
    "id": 119,
    "process": "prover",
    "message": "Wrap_verifier_incrementally_verify_proof_done",
    "metadata": [],
    "timestamp": "2023-04-27 00:59:08.134230Z"
}
```
To get `snarky_js_bindings` to build, the `logger` library is split into `native` and `fake` implementations, because the native logger pulls in `core` and `async`, in order to use the `Async.Rpc` mechanism.

This PR also deletes the client SDK and attendant CI tasks. There are still some references to the client SDK in `automation` and `nix`. The client SDK uses the logger. It could have been made to work with the fake logger implementation, but it's been replaced by the native `mina-signer` library, so not worth the effort to preserve it.

Closes #13092.
Closes https://github.com/o1-labs/snarkyjs/issues/837

### CODE REVIEW

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

File: src/lib/mina_lib/mina_lib.ml [hunk 1]
```diff
@@ -1434,13 +1456,35 @@ let create ?wallets (config : Config.t) =
...
+                        | Ok () ->
+                            ()
+                        | Error err ->
+                            [%log' warn config.logger]
+                              "Error when setting ITN logger data for \
+                               verifier: %s"
+                              (Error.to_string_hum err)
+                      else Deferred.unit
+                    in
+                    verifier ) )
```
Comment (Sventimir) [inline src/lib/mina_lib/mina_lib.ml:1487]: This code appears twice and the appearences seem to only differ by the module (`Prover` vs. `Verifier`), both of which seem to conveniently have the same signature to the extent required here. Let's extract this patch and make it a function!

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Review by jdsteinhauser — APPROVED** [review]

**Comment by psteckler** [comment]
!ci-build-me

**Comment by mitschabaude** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Review by stevenplatt — APPROVED** [review]

File: src/lib/logger/native/logger.ml [hunk 2]
```diff
@@ -371,18 +363,17 @@ let log t ~level ~module_ ~location ?tags ?(metadata = []) ?event_id fmt =
...
         ()
   in
-
   ksprintf f fmt
 
 type 'a log_function =
      t
   -> module_:string
   -> location:string
-  -> ?tags:Tags.t list
```
Comment (deepthiskumar) [inline src/lib/logger/native/logger.ml:385]: Sad that we never got to using this. Is it still worth keeping it though?

**Review by deepthiskumar — APPROVED** [review]

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**